libs/kameleon/src/kameleon/app_listing.clj

```diff
@@ -309,3 +309,32 @@
   (-> (get-app-listing-base-query workspace favorites-group-index query-opts)
       (add-public-apps-by-user-where-clause email)
       (select)))
+
+(defn- is-visible-app-subselect
+  [{workspace-id :id root-category-id :root_category_id} favorites-group-index app-id-keyword]
+  (subselect [:app_category_app :aca]
+             (join [:app_category_listing :ac] {:aca.app_category_id :ac.id})
+             (where {:aca.app_id app-id-keyword})
+             (where {:ac.id [not= (get-fav-group-id-subselect root-category-id favorites-group-index)]})
+             (where (or :ac.is_public {:ac.workspace_id workspace-id}))))
+
+(defn list-shared-apps
+  "Lists apps that have been shared with a user. For the time being, this works by listing all apps
+  in th :app-ids parametr that are not in one of the categories that the user can access. When the
```

Comment:
nearly irrelevantly small but may as well spell out 'the' and 'parameter' with all their vowels